packages/fast-tooling-react/src/form/form/form-item.dictionary.tsx

```diff
@@ -0,0 +1,229 @@
+                    "formItemDictionary_controlRegion"
+                )}
+            >
+                <div
+                    className={get(
+                        this.props.managedClasses,
+                        "formItemDictionary_control"
+                    )}
+                >
+                    <label
+                        className={get(
+                            this.props.managedClasses,
+                            "formItemDictionary_controlLabel"
+                        )}
+                    >
+                        {this.props.label}
+                    </label>
+                </div>
+                <button
+                    className={get(
+                        this.props.managedClasses,
+                        "formItemDictionary_controlAddTrigger"
+                    )}
+                    aria-label={"Select to add item"}
+                    onClick={this.handleOnAddItem}
+                />
+            </div>
+        );
+    }
+
+    private renderItemControl(propertyName: string): React.ReactNode {
+        return (
+            <div
+                className={get(
+                    this.props.managedClasses,
+                    "formItemDictionary_itemControlRegion"
+                )}
+            >
+                <div
+                    className={get(
+                        this.props.managedClasses,
+                        "formItemDictionary_itemControl"
+                    )}
+                >
+                    <label
+                        className={get(
+                            this.props.managedClasses,
+                            "formItemDictionary_itemControlLabel"
+                        )}
+                    >
+                        {this.props.schema.propertyTitle || "Property key"}
+                    </label>
+                    <input
+                        className={get(
+                            this.props.managedClasses,
+                            "formItemDictionary_itemControlInput"
+                        )}
+                        type="text"
+                        value={propertyName}
+                        onChange={this.handleOnKeyChange(propertyName)}
+                    />
+                    <button
+                        className={get(
+                            this.props.managedClasses,
+                            "formItemDictionary_itemControlRemoveTrigger"
+                        )}
+                        onClick={this.handleOnRemoveItem(propertyName)}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    private renderFormControls(): React.ReactNode {
+        return (typeof this.props.data !== "undefined"
+            ? Object.keys(this.props.data)
+            : []
+        )
+            .filter((dataKey: string) => {
+                return !this.props.enumeratedProperties.includes(dataKey);
+            })
```

Comment:
You could just use reduce and avoid multiple loops on the array.

just saying...

Reply:
Darn it